### PR TITLE
feat: Manage private packages upload for Gatling Enterprise, closes HYB-253

### PR DESCRIPTION
**Pull request by Obero**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 target/
 version.sbt
+/.idea/
+/.bsp/
```
Comment (Isammoc) [inline .gitignore:4]: `.idea` is linked to your choice to use IntelliJ Idea as IDE.
So, it should go in your own global gitignore ([blog article](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/))

For `.bsp`...  can be acceptable (as it is produced by sbt)

File: src/main/java/io/gatling/plugin/client/http/PrivatePackagesApiRequests.java
Comment (Isammoc) [inline src/main/java/io/gatling/plugin/client/http/PrivatePackagesApiRequests.java:1]: Please :pray: 
Avoid default visibility (package visibility) when possible.

File: .gitignore (hunk 1 above)
Comment (Obero) [reply .gitignore:4]: Agreed :)
A bit of cleaning should be made in other repos to match this point of view, I'll take a look when passing by.

Comment (Isammoc) [reply .gitignore:4]: Take care that demo repositories does not follow the same conventions (as a lot of open source project tend to configure common IDE gitignore...)

File: src/main/java/io/gatling/plugin/client/http/PrivatePackagesApiRequests.java
Comment (Obero) [reply src/main/java/io/gatling/plugin/client/http/PrivatePackagesApiRequests.java:1]: WDYM ?
I prefer to use the most restrictive access level, and package private matches this criterion (cannot be private).

Comment (Isammoc) [reply src/main/java/io/gatling/plugin/client/http/PrivatePackagesApiRequests.java:1]: At every level:
 * class (maybe this one is needed for your use case)
 * constructor (if the class is package private, the constructor may be public)
 * and so on

Comment (Obero) [reply src/main/java/io/gatling/plugin/client/http/PrivatePackagesApiRequests.java:1]: I'm afraid I don't understand what bothers you...

- The class visibility is the most restrictive one I can use, which seems ok to me
- The constructor has no reason to have a broader visibility than its class
- Same thing for the method
- I don't have a use case needing a broader access

And this is consistent with other `...ApiRequests` classes.

Is this a coding style question ?

**Review by Isammoc — APPROVED** [review]

**Review by tpetillot — APPROVED** [review]

**Pull request merged**